tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -36,14 +38,22 @@ internal static class DatabaseMonitoringPropagator
 
         private static int _remainingErrorLogs = 100; // to prevent too many similar errors in the logs. We assume that after 100 logs, the incremental value of more logs is negligible.
 
-        internal static bool PropagateDataViaComment(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbCommand command, string configuredServiceName, string? dbName, string? outhost, Span span)
+        internal static bool PropagateDataViaComment(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbCommand command, string configuredServiceName, string? dbName, string? outhost, Span span, bool injectStoredProcedure)
         {
             if (integrationId is not (IntegrationId.MySql or IntegrationId.Npgsql or IntegrationId.SqlClient or IntegrationId.Oracle) ||
                 propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full))
             {
                 return false;
             }
 
+            if ((!injectStoredProcedure && command.CommandType == CommandType.StoredProcedure)
+                || (command.CommandType == CommandType.StoredProcedure && integrationId != IntegrationId.SqlClient))
```

Comment:
Sorry for being annoying again 😄

I like it like that, I think we could simplify this with 
```Csharp 
if (command.CommandType == CommandType.StoredProcedure && (!injectStoredProcedure || integrationId != IntegrationId.SqlClient))
```

Feel free to add your own flavour to it, I'm just giving ideas to make it more readable 😉

Reply:
Thanks! Applied :)